Mathlib/Analysis/NormedSpace/Alternating/Basic.lean

```diff
@@ -0,0 +1,599 @@
+    E [⋀^ι]→L[𝕜] (F [⋀^ι']→L[𝕜] G)  :=
+  mkContinuous
+    { toFun m := mkContinuous (f m) (C * ∏ i, ‖m i‖) <| H m
+      map_update_add' m i x y := by ext1; simp
+      map_update_smul' m i c x := by ext1; simp
+      map_eq_zero_of_eq' v i j hv hij := by
+        ext v'
+        have : f v = 0 := by simpa using f.map_eq_zero_of_eq' v i j hv hij
+        simp [this] }
+    (max C 0) fun m => by
+      simp only [coe_mk, MultilinearMap.coe_mk]
+      refine ((f m).mkContinuous_norm_le' _).trans_eq ?_
+      rw [max_mul_of_nonneg, zero_mul]
+      positivity
+
+@[simp]
+theorem mkContinuousAlternating_apply (f : E [⋀^ι]→ₗ[𝕜] (F [⋀^ι']→ₗ[𝕜] G)) {C : ℝ}
+    (H : ∀ m₁ m₂, ‖f m₁ m₂‖ ≤ (C * ∏ i, ‖m₁ i‖) * ∏ i, ‖m₂ i‖) (m : ι → E) :
+    ⇑(mkContinuousAlternating f C H m) = f m :=
+  rfl
+
+theorem mkContinuousAlternating_norm_le_max (f : E [⋀^ι]→ₗ[𝕜] (F [⋀^ι']→ₗ[𝕜] G)) {C : ℝ}
+    (H : ∀ m₁ m₂, ‖f m₁ m₂‖ ≤ (C * ∏ i, ‖m₁ i‖) * ∏ i, ‖m₂ i‖) :
+    ‖mkContinuousAlternating f C H‖ ≤ max C 0 := by
+  dsimp only [mkContinuousAlternating]
+  exact mkContinuous_norm_le _ (le_max_right _ _) _
+
+theorem mkContinuousAlternating_norm_le (f : E [⋀^ι]→ₗ[𝕜] (F [⋀^ι']→ₗ[𝕜] G)) {C : ℝ}
+    (hC : 0 ≤ C) (H : ∀ m₁ m₂, ‖f m₁ m₂‖ ≤ (C * ∏ i, ‖m₁ i‖) * ∏ i, ‖m₂ i‖) :
+    ‖mkContinuousAlternating f C H‖ ≤ C :=
+  (mkContinuousAlternating_norm_le_max f H).trans_eq (max_eq_left hC)
+
+end AlternatingMap
+
+end Seminorm
+
+section Norm
+
+/-! Results that are only true if the target space is a `NormedAddCommGroup`
+(and not just a `SeminormedAddCommGroup`). -/
+
+universe u wE wF v
+variable {𝕜 : Type u} {n : ℕ} {E : Type wE} {F : Type wF} {ι : Type v}
+  [Fintype ι]
+  [NontriviallyNormedField 𝕜]
+  [SeminormedAddCommGroup E] [NormedSpace 𝕜 E]
+  [NormedAddCommGroup F] [NormedSpace 𝕜 F]
+
+namespace ContinuousAlternatingMap
+
+/-- Continuous alternating maps themselves form a normed group with respect to
+    the operator norm. -/
+instance instNormedAddCommGroup : NormedAddCommGroup (E [⋀^ι]→L[𝕜] F) :=
+  NormedAddCommGroup.ofSeparation fun _f hf ↦
+    toContinuousMultilinearMap_injective <| norm_eq_zero.mp hf
+
+variable (𝕜 F) in
+theorem norm_ofSubsingleton_id [Subsingleton ι] [Nontrivial F] (i : ι) :
+    ‖ofSubsingleton 𝕜 F F i (.id _ _)‖ = 1 :=
+  ContinuousMultilinearMap.norm_ofSubsingleton_id 𝕜 F i
+
+variable (𝕜 F) in
+theorem nnnorm_ofSubsingleton_id [Subsingleton ι] [Nontrivial F] (i : ι) :
+    ‖ofSubsingleton 𝕜 F F i (.id _ _)‖₊ = 1 :=
+  NNReal.eq <| norm_ofSubsingleton_id ..
+
+end ContinuousAlternatingMap
+
+namespace AlternatingMap
+
+/-- If an alternating map in finitely many variables on a normed space satisfies the inequality
+`‖f m‖ ≤ C * ∏ i, ‖m i‖` on a shell `ε i / ‖c i‖ < ‖m i‖ < ε i` for some positive numbers `ε i`
+and elements `c i : 𝕜`, `1 < ‖c i‖`, then it satisfies this inequality for all `m`. -/
+theorem bound_of_shell (f : F [⋀^ι]→L[𝕜] E) {ε : ι → ℝ} {C : ℝ} {c : ι → 𝕜}
```

Comment:
Shouldn't `f` be just multilinear (instead of continuous multilinear)?

Reply:
Thanks! Fixed.